src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -100,25 +104,98 @@ public async Task<bool> WaitForMergedPullRequestAsync(string targetRepo, string
             throw new MaestroTestException($"The created pull request for {targetRepo} targeting {targetBranch} was not merged within {attempts} minutes");
         }
 
-        private async Task<Microsoft.DotNet.DarcLib.PullRequest> WaitForAzDoPullRequestAsync(string targetRepoUri, string targetBranch)
+        private async Task<int> GetAzDoPullRequestIdAsync(string targetRepoName, string targetBranch)
         {
+            string searchBaseUrl = GetAzDoRepoUrl(targetRepoName);
+            string apiBaseUrl = GetAzDoApiRepoUrl(targetRepoName);
+            IEnumerable<int> prs = new List<int>();
+
             int attempts = 10;
             while (attempts-- > 0)
             {
-                var prs = await AzDoClient.SearchPullRequestsAsync(targetRepoUri, targetBranch, Microsoft.DotNet.DarcLib.PrStatus.Open).ConfigureAwait(false);
+                try
+                {
+                    prs = await SearchPullRequestsAsync(searchBaseUrl, targetBranch).ConfigureAwait(false);
+                }
+                catch (HttpRequestException ex)
+                {
+                    if (ex.Message == "Response status code does not indicate success: 404 (Not Found).")
```

Comment:
Reading the message, especially the full message, is a bit fragile.  This isn't really your fault... the method we've created doesn't return enough information to be called usefully. :-( I'd just retry everything (remove the "if" and just swallow everything in the catch, probably logging something to TestContext/ILogger)